Serce gildii powoduje problemy przy spawnowaniu asynchronicznie

```
[00:53:43] [Craft Scheduler Thread - 0/ERROR]: stopTiming called async for world - chunkLoad - Entities
[00:53:43] [Craft Scheduler Thread - 0/WARN]: java.lang.Throwable
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at co.aikar.timings.TimingHandler.stopTiming(TimingHandler.java:114)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at net.minecraft.server.v1_8_R3.ChunkRegionLoader.loadEntities(ChunkRegionLoader.java:424)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.chunkio.ChunkIOProvider.callStage2(ChunkIOProvider.java:46)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.chunkio.ChunkIOProvider.callStage2(ChunkIOProvider.java:1)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.util.AsynchronousExecutor.skipQueue(AsynchronousExecutor.java:337)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.util.AsynchronousExecutor.getSkipQueue(AsynchronousExecutor.java:295)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.chunkio.ChunkIOExecutor.syncChunkLoad(ChunkIOExecutor.java:16)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at net.minecraft.server.v1_8_R3.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:135)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at net.minecraft.server.v1_8_R3.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:117)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.CraftWorld.getChunkAt(CraftWorld.java:145)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.CraftWorld.getChunkAt(CraftWorld.java:604)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.Location.getChunk(Location.java:78)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at net.dzikoysk.funnyguilds.guild.RegionManager.findRegionAtLocation(RegionManager.java:92)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at net.dzikoysk.funnyguilds.listener.region.PlayerMove.lambda$onMove$6(PlayerMove.java:50)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftTask.run(CraftTask.java:59)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:53)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.github.paperspigot.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:23)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at java.lang.Thread.run(Thread.java:748)
[00:53:43] [Craft Scheduler Thread - 0/ERROR]: stopTiming called async for world - chunkLoad - TileEntities
[00:53:43] [Craft Scheduler Thread - 0/WARN]: java.lang.Throwable
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at co.aikar.timings.TimingHandler.stopTiming(TimingHandler.java:114)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at net.minecraft.server.v1_8_R3.ChunkRegionLoader.loadEntities(ChunkRegionLoader.java:438)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.chunkio.ChunkIOProvider.callStage2(ChunkIOProvider.java:46)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.chunkio.ChunkIOProvider.callStage2(ChunkIOProvider.java:1)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.util.AsynchronousExecutor.skipQueue(AsynchronousExecutor.java:337)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.util.AsynchronousExecutor.getSkipQueue(AsynchronousExecutor.java:295)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.chunkio.ChunkIOExecutor.syncChunkLoad(ChunkIOExecutor.java:16)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at net.minecraft.server.v1_8_R3.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:135)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at net.minecraft.server.v1_8_R3.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:117)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.CraftWorld.getChunkAt(CraftWorld.java:145)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.CraftWorld.getChunkAt(CraftWorld.java:604)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.Location.getChunk(Location.java:78)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at net.dzikoysk.funnyguilds.guild.RegionManager.findRegionAtLocation(RegionManager.java:92)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at net.dzikoysk.funnyguilds.listener.region.PlayerMove.lambda$onMove$6(PlayerMove.java:50)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftTask.run(CraftTask.java:59)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:53)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.github.paperspigot.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:23)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at java.lang.Thread.run(Thread.java:748)
[00:53:43] [Craft Scheduler Thread - 0/ERROR]: stopTiming called async for world - chunkLoad - TileTicks
[00:53:43] [Craft Scheduler Thread - 0/WARN]: java.lang.Throwable
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at co.aikar.timings.TimingHandler.stopTiming(TimingHandler.java:114)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at net.minecraft.server.v1_8_R3.ChunkRegionLoader.loadEntities(ChunkRegionLoader.java:459)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.chunkio.ChunkIOProvider.callStage2(ChunkIOProvider.java:46)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.chunkio.ChunkIOProvider.callStage2(ChunkIOProvider.java:1)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.util.AsynchronousExecutor.skipQueue(AsynchronousExecutor.java:337)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.util.AsynchronousExecutor.getSkipQueue(AsynchronousExecutor.java:295)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.chunkio.ChunkIOExecutor.syncChunkLoad(ChunkIOExecutor.java:16)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at net.minecraft.server.v1_8_R3.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:135)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at net.minecraft.server.v1_8_R3.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:117)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.CraftWorld.getChunkAt(CraftWorld.java:145)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.CraftWorld.getChunkAt(CraftWorld.java:604)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.Location.getChunk(Location.java:78)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at net.dzikoysk.funnyguilds.guild.RegionManager.findRegionAtLocation(RegionManager.java:92)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at net.dzikoysk.funnyguilds.listener.region.PlayerMove.lambda$onMove$6(PlayerMove.java:50)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftTask.run(CraftTask.java:59)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:53)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at org.github.paperspigot.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:23)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: 	at java.lang.Thread.run(Thread.java:748)
[00:53:43] [Craft Scheduler Thread - 0/WARN]: Exception in thread "Craft Scheduler Thread - 0" 
[00:53:43] [Craft Scheduler Thread - 0/WARN]: org.apache.commons.lang.UnhandledException: Plugin FunnyGuilds v4.12.1-SNAPSHOT Snowdrop-c62d24d generated an exception while executing task 287
	at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:56)
	at org.github.paperspigot.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:23)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalStateException: Asynchronous entity world add!
	at org.spigotmc.AsyncCatcher.catchOp(AsyncCatcher.java:14)
	at net.minecraft.server.v1_8_R3.World.b(World.java:2738)
	at net.minecraft.server.v1_8_R3.Chunk.addEntities(Chunk.java:912)
	at org.bukkit.craftbukkit.v1_8_R3.chunkio.ChunkIOProvider.callStage2(ChunkIOProvider.java:49)
	at org.bukkit.craftbukkit.v1_8_R3.chunkio.ChunkIOProvider.callStage2(ChunkIOProvider.java:1)
	at org.bukkit.craftbukkit.v1_8_R3.util.AsynchronousExecutor.skipQueue(AsynchronousExecutor.java:337)
	at org.bukkit.craftbukkit.v1_8_R3.util.AsynchronousExecutor.getSkipQueue(AsynchronousExecutor.java:295)
	at org.bukkit.craftbukkit.v1_8_R3.chunkio.ChunkIOExecutor.syncChunkLoad(ChunkIOExecutor.java:16)
	at net.minecraft.server.v1_8_R3.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:135)
	at net.minecraft.server.v1_8_R3.ChunkProviderServer.getChunkAt(ChunkProviderServer.java:117)
	at org.bukkit.craftbukkit.v1_8_R3.CraftWorld.getChunkAt(CraftWorld.java:145)
	at org.bukkit.craftbukkit.v1_8_R3.CraftWorld.getChunkAt(CraftWorld.java:604)
	at org.bukkit.Location.getChunk(Location.java:78)
	at net.dzikoysk.funnyguilds.guild.RegionManager.findRegionAtLocation(RegionManager.java:92)
	at net.dzikoysk.funnyguilds.listener.region.PlayerMove.lambda$onMove$6(PlayerMove.java:50)
	at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftTask.run(CraftTask.java:59)
	at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:53)
	... 4 more
```

--- a/plugin/src/main/java/net/dzikoysk/funnyguilds/guild/RegionManager.java
+++ b/plugin/src/main/java/net/dzikoysk/funnyguilds/guild/RegionManager.java
@@ -89,7 +89,8 @@ public class RegionManager {
      * @return the region
      */
     public Option<Region> findRegionAtLocation(Location location) {
-        Set<Region> applicableRegions = this.regions.get(packChunkPosition(location.getChunk().getX(), location.getChunk().getZ()));
+        long packedPos = packChunkPosition(location.getBlockX() / 16, location.getBlockZ() / 16);
+        Set<Region> applicableRegions = this.regions.get(packedPos);
 
         if (applicableRegions == null || applicableRegions.isEmpty()) {
             return Option.none();

--- a/plugin/src/main/java/net/dzikoysk/funnyguilds/listener/region/PlayerMove.java
+++ b/plugin/src/main/java/net/dzikoysk/funnyguilds/listener/region/PlayerMove.java
@@ -84,9 +84,7 @@ public class PlayerMove extends AbstractFunnyListener {
                         }
 
                         if (this.config.heart.createEntityType != null) {
-                            Bukkit.getScheduler().runTaskLaterAsynchronously(this.plugin, () -> {
-                                this.guildEntityHelper.spawnGuildEntity(guild, player);
-                            }, 40L);
+                            Bukkit.getScheduler().runTaskLater(this.plugin, () -> this.guildEntityHelper.spawnGuildEntity(guild, player), 40L);
                         }
 
                         FunnyFormatter formatter = new FunnyFormatter()

